src/components/ImageView/index.js

```diff
@@ -197,14 +203,15 @@ class ImageView extends PureComponent {
             return (
                 <View
                     style={[styles.imageViewContainer, styles.overflowHidden]}
+                    onLayout={this.onContainerLayoutChanged}
                 >
                     <Image
                         source={{uri: this.props.url}}
-                        style={[
+                        style={this.state.zoomScale === 0 ? undefined : [
                             styles.w100,
                             styles.h100,
-                        ]}
-                        resizeMode="contain"
+                        ]} // Hide image until zoomScale calculated
+                        resizeMode={this.state.zoomScale > 1 ? 'center' : 'contain'} // For big dimension images 'center' makes resizing the view. There is no sizing with 'contain'.
```

Comment:
I think we can rely on `contain` for now. But change the comment to 
```suggestion
                        resizeMode={this.state.zoomScale > 1 ? 'center' : 'contain'} // When Image dimensions are lower than the container boundary(zoomscale <= 1), use `contain` to render the image with natural dimensions. Both `center` and `contain` keeps the image centered on both x and y axis.
```